### DISCUSSION TITLE
Search for dependencies/versions

### DISCUSSION THREAD

**Discussion by haesleinhuepf** [chanzuckerberg/napari-hub#581] [Ideas]
Hi all,

I'd like to suggest a feature: Searching plugins that are compatible with specific dependencies/versions and/or napari versions. When searching for "napari==0.4.15" for example, this finds two plugins, but none of those specify this dependency:

![image](https://user-images.githubusercontent.com/12660498/174928353-3f3dbf7a-297b-40f0-8998-7cf0e989c064.png)

However, there are plugins that are explicitly pinned to this version and it would be cool if those are discoverable using the search:

![image](https://user-images.githubusercontent.com/12660498/174928543-566fc6f4-27ec-4ddb-98e7-45c5d3f62871.png)

Btw searching for "napari (==0.4.15)" delivers no result.

![image](https://user-images.githubusercontent.com/12660498/174928585-6f86876b-eb5d-4efe-9099-e67b29e61cfc.png)

I presume searching for dependencies was never supported and thus, this might be no bug per definition. Let's treat it as a feature request. :-)

If we had that feature, we could also search for "tensorflow", "pytorch" or any other dependency, and see what versions other plugins support. That could help the community to make plugins compatible with each other.

Looking forward to hear what others think!

**Comment by neuromusic** [comment]
Thanks for the idea @haesleinhuepf :D 

A few clarifications...

The key motivation for this would be answering "which plugins depend on <python package>?" and "which plugins depend on <a given version of a Python package>?" is that right?

The first one could be addressed by a filter or indexing the "requirements" to search... however, we would be limited to the _explicit_ dependencies and would miss any transitive dependencies.

The second Q would further need some additional logic w/r/t semver (similar to what we do for Python version requirements)... a simple search or filter based on the info the plugin author contributes would be inadequate I think.
